qcodes/tests/dataset/dond/test_doNd.py

```diff
@@ -1863,3 +910,641 @@ def test_dond_2d_multi_datasets_with_callable_output_data(
+
+    output = dond(together_sweep, [_param], [_param_2], do_plot=False)
+
+    assert sweep_1.param.name == "simple_setter_parameter"
+    assert _param.name == "simple_parameter"
+
+    assert sweep_2.param.name == "simple_setter_parameter_2"
+    assert _param_2.name == "simple_parameter_2"
+
+    assert (
+        output[0][0].parameters
+        == "simple_setter_parameter,simple_setter_parameter_2,simple_parameter"
+    )
+    assert (
+        output[0][1].parameters
+        == "simple_setter_parameter,simple_setter_parameter_2,simple_parameter_2"
+    )
+
+
+@pytest.mark.usefixtures("plot_close", "experiment")
+def test_dond_together_sweep_sweeper(_param_set, _param_set_2, _param):
+
+    sweep_len = 10
+
+    delay_1 = 0.1
+
+    sweep_1 = LinSweep(_param_set, 0, 1, sweep_len, delay_1)
+
+    delay_2 = 0.2
+
+    sweep_2 = LinSweep(_param_set_2, 1, 2, sweep_len, delay_2)
+
+    together_sweep = TogetherSweep(sweep_1, sweep_2)
+
+    sweeper = _Sweeper([together_sweep], [])
+
+    expected_sweeper_groups = (
+        (sweep_1.param,),
+        (sweep_2.param,),
+        (
+            sweep_1.param,
+            sweep_2.param,
+        ),
+    )
+    for g in expected_sweeper_groups:
+        assert g in sweeper.sweep_groupes
+    assert len(expected_sweeper_groups) == len(sweeper.sweep_groupes)
+    assert sweeper.shape == (10,)
+    assert sweeper.all_setpoint_params == (sweep_1.param, sweep_2.param)
+
+    assert list(sweeper.setpoint_dict.keys()) == [
+        "simple_setter_parameter",
+        "simple_setter_parameter_2",
+    ]
+    assert sweeper.setpoint_dict["simple_setter_parameter"] == list(
+        sweep_1.get_setpoints()
+    )
+    assert sweeper.setpoint_dict["simple_setter_parameter_2"] == list(
+        sweep_2.get_setpoints()
+    )
+
+    for output, setpoint_1, setpoint_2 in zip(
+        sweeper, sweep_1.get_setpoints(), sweep_2.get_setpoints()
+    ):
+        assert output[0].parameter == sweep_1.param
+        assert output[0].new_value == setpoint_1
+        assert output[0].should_set is True
+        assert output[0].delay == delay_1
+
+        assert output[1].parameter == sweep_2.param
+        assert output[1].new_value == setpoint_2
+        assert output[1].should_set is True
+        assert output[1].delay == delay_2
+
+
+def test_dond_together_sweep_sweeper_combined():
+    a = ManualParameter("a", initial_value=0)
+    b = ManualParameter("b", initial_value=0)
+    c = ManualParameter("c", initial_value=0)
+    d = ManualParameter("d", initial_value=1)
+    e = ManualParameter("e", initial_value=2)
+    f = ManualParameter("f", initial_value=3)
+    sweepA = LinSweep(a, 0, 3, 10)
+    sweepB = LinSweep(b, 5, 7, 10)
+    sweepC = LinSweep(c, 8, 12, 10)
+
+    datasets, _, _ = dond(
+        TogetherSweep(sweepA, sweepB),
+        sweepC,
+        d,
+        e,
+        f,
+        do_plot=False,
+        dataset_dependencies={
+            "ds1": (a, c, d),
```

Comment:
i didn't see a test where more than one measured parameter is added to a dataset, e.g. `(a,c, e,f,)` - can we add such a test for completeness?